code/game/objects/structures/emergency_dispenser.dm

```diff
@@ -0,0 +1,39 @@
+
+//spawns endless (3 sets) amounts of breathmask, emergency oxy tank
+/obj/structure/emergency_dispenser
+	name = "emergency dispenser"
+	desc = "A wall mounted dispenser with emergency supplies."
+	icon = 'icons/obj/structures/emergency_dispenser.dmi'
+	icon_state = "world"
+	anchored = TRUE
+	var/static/list/spawnitems = list(/obj/item/tank/emergency/oxygen,/obj/item/clothing/mask/breath)
+	var/amount = 3 // spawns each items X times.
+
+/obj/structure/emergency_dispenser/attack_hand(mob/user)
+	if(!CanPhysicallyInteract(user))	//Added by Strumpetplaya - AI shouldn't be able to         (you're both stupid, need CanPhysicallyInteract --Chinsky)
+		return									//activate emergency lockers.  This fixes that.  (Does this make sense, the AI can't call attack_hand, can it? --Mloc) 
+	if(!amount)
+		to_chat(user, SPAN_WARNING("It's empty."))
+		return
```

Comment:
`return TRUE` to indicate the action completed.

Reply:
donezo